_episodes/01-Basic_OOP.md

```diff
@@ -0,0 +1,306 @@
+
+```
+'a' is an instance of the class <class '__main__.MyClass'>
+'b' is an instance of the class <class '__main__.MyClass'>
+Are 'a' and 'b' equal? False
+```
+{: .output}
+
+> ## Built-in classes (or types)
+> Python built-in types like `list`, `tuple`, `dict`, `str` and all the numeric types are
+> classes, and the variables that use them are instances of those classes. To what classes
+> belong the following variables?
+>
+> ```python
+> a = [1, 2, 3]
+> b = (1, 2, 3)
+> c = {1, 2, 3}
+> d = 42
+> e = {"name": "Fluffy", "species": "Cat"}
+> f = True
+> g = "My taylor is rich!"
+> ```
+> > ## Solution
+> > Using `print(type(variable_name))` in each case results in:
+> > - a: `<class 'list'>`
+> > - b: `<class 'tuple'>`
+> > - c: `<class 'set'>`
+> > - d: `<class 'int'>`
+> > - e: `<class 'dict'>`
+> > - f: `<class 'bool'>`
+> > - g: `<class 'str'>`
+> {: .solution}
+{: .challenge}
+
+## Methods
+
+- The above example class has no attributes nor methods, which is pretty useless.
+- When creating a class, you must define also its attributes and methods, so the rest of the code can interact with it.
+- Methods are the behaviors of the class, the functions or actions that can be used to interact with the class.
+- They are defined like any other function in Python, using the `def` keyword, but with a first input parameter that has a special meaning.
+
+```python
+class MySecondClass:
+    """ A simple example class with a method. """
+
+    def say_hello(self, name):
+        print(f"Hello {name}! This instance is {self}")
+```
+
+- Here, `MySecondClass` has a method, `say_hello` with two input parameters:
+    - The first one, usually called `self` refers to the specific instance of this class being used. It is compulsory: all methods within a class must take this parameter as first argument.
+    - The second argument is just a parameter used within the method.
+- As with any function in Python, methods can have any number of positional and keyword arguments, as well as returning values using the keyword `return`.
+- To call a method of a class, use the dot notation `.` to separate the name of the object and the method being invoked.
+- The value for the first argument, `self`, is assigned automatically to the instance being used and **must not be provided**:
+
+```python
+a = MySecondClass()
+b = MySecondClass()
+
+a.say_hello("Diego")
+b.say_hello("Diego")
+```
+```
+Hello Diego! This instance is <__main__.MySecondClass object at 0x10ef01780>
+Hello Diego! This instance is <__main__.MySecondClass object at 0x10ef01be0>
+```
+{: .output}
+
+- As you can see by the numbers at the end, `a` and `b` are two different instances of class `MySecondClass`.
+
+## Attributes
+
+- They are the properties of the class, the variables that can be read and modified (usually).
+- There are two types of attributes: **class attributes** and **instance attributes**.
+
+### Class attributes
```

Comment:
Is this necessary information? How often do you use class attributes? I tend to avoid them (or at least avoid the mutable data types), because of the risk of editing an object I wasn't intending on editing.

Reply:
I'm moving the class atributes to a separate `callout` section, possibly at the end. I don't use them very much, but I think they are often useful, so they should know they exist.